src/main/kotlin/task/GithubReleaseTasks.kt

```diff
@@ -46,139 +43,15 @@ class GithubReleaseTaskException(override var message: String,
-    val notConfigured = GithubReleaseTaskException(
-        """No github user name configured.
-          |Configure it
-          |- with the project property $CONFIG_OPT_GITHUB_USER
-          |- or with the environment variable $ENV_VAR_GITHUB_USER"""
-            .trimMargin("|")
-    )
-
-    return this.lookupConfiguredProperty(
-        propertyName = CONFIG_OPT_GITHUB_USER,
-        envName = ENV_VAR_GITHUB_USER
-    ) ?: throw notConfigured
-}
-
-val Project.configuredGithubAccessToken: String get() {
-    val notConfigured = GithubReleaseTaskException(
-        """No github access token configured.
-            |Configure it
-            |- with the project property $CONFIG_OPT_GITHUB_ACCESS_TOKEN
-            |- or with the environment variable $ENV_VAR_GITHUB_ACCESS_TOKEN"""
-            .trimMargin("|")
-    )
-
-    return lookupConfiguredProperty(
-        propertyName =  CONFIG_OPT_GITHUB_ACCESS_TOKEN,
-        envName = ENV_VAR_GITHUB_ACCESS_TOKEN
-    ) ?: throw notConfigured
-}
-
-val Project.defaultReleasesConfigFile: File get() {
-    return File(this.project.projectDir, "releases.yml")
-}
-
-fun Project.releaseUrl(label: String) =
-    "${this.configuredGithubUrl}/${this.configuredGithubUser}/" +
-        "${this.configuredGithubRepository}/releases/tag/$label"
-
-/**
- * the configured release config file
- */
-val Project.configuredReleasesConfigFile: File get() {
-    fun ensureReleaseConfigFileReadable(configFile: File) {
-        if (! (configFile.isFile && configFile.exists()
-                && configFile.canRead())) {
-            throw IOException(
-                "releases configuration file <${configFile.absolutePath}> "
-                    + "doesn't exist or can't be read")
-        }
-    }
-
-    val file =
-        lookupConfiguredProperty(
-            propertyName = CONFIG_OPT_RELEASES_CONFIG_FILE
-        )?.trim().let {name ->
-            if (name.isNullOrEmpty()) null else File(name)
-        } ?: defaultReleasesConfigFile
-
-    ensureReleaseConfigFileReadable(file)
-    return file
-}
-
-val Project.configuredGithubApiUrl: String get () {
-    return lookupConfiguredProperty(
-            propertyName = CONFIG_OPT_GITHUB_API_URL,
-            envName = ENV_VAR_GITHUB_API_URL
-    ) ?: DEFAULT_GITHUB_API_URL
-}
-
-val Project.configuredGithubUploadUrl: String get () {
-    return lookupConfiguredProperty(
-        propertyName = CONFIG_OPT_GITHUB_UPLOAD_URL,
-        envName = ENV_VAR_GITHUB_UPLOAD_URL
-    ) ?: DEFAULT_GITHUB_UPLOAD_URL
-}
-
-val Project.configuredGithubUrl: String get () {
-    return lookupConfiguredProperty(
-        propertyName = CONFIG_OPT_GITHUB_URL,
-        envName = ENV_VAR_GITHUB_URL
-    ) ?: DEFAULT_GITHUB_URL
-}
-
-val Project.configuredGithubRepository: String get () {
-    return lookupConfiguredProperty(
-        propertyName = CONFIG_OPT_GITHUB_REPOSITORY,
-        envName = ENV_VAR_GITHUB_REPOSITORY
-    ) ?: project.name
```

Comment:
I dropped the fallback to the project name, since Gradle derives the project name from the project directory, which can be different between developers. Better state the repository name in the `build.gradle` explicitly.